src/shogun/lib/SGVector.cpp

```diff
@@ -309,6 +309,15 @@ void SGVector<T>::resize_vector(int32_t n)
 	vlen=n;
 }
 
+template <class T>
+SGVector<T> SGVector<T>::slice(index_t l, index_t h) const
+{
+	assert_on_cpu();
+	ASSERT(l >= 0 && h <= vlen);
+
+	return SGVector<T>(vector, h - l, l);
```

Comment:
so you do reference counting for the returned vector?

Reply:
no reference counting